applications/shape_detector/src/plane_detector.py

```diff
@@ -0,0 +1,939 @@
+log.getLogger('matplotlib').setLevel(log.WARNING)
+log.getLogger('PIL').setLevel(log.WARNING)
+
+import matplotlib.pyplot as plt
+
+#%% Helpers
+def draw_axis(img, rvec, tvec, cam_mtrx, cam_dist, len = 10):
+    # unit is mm
+    points          = np.float32([[len, 0, 0], [0, len, 0], [0, 0, len], [0, 0, 0]]).reshape(-1, 3)
+    axisPoints, _   = cv.projectPoints(points, rvec, tvec, cam_mtrx, cam_dist)
+    axisPoints      = axisPoints.squeeze().astype(np.int32)
+    img = cv.line(img, tuple(axisPoints[3].ravel()), tuple(axisPoints[0].ravel()), (255,0,0), 3)
+    img = cv.line(img, tuple(axisPoints[3].ravel()), tuple(axisPoints[1].ravel()), (0,255,0), 3)
+    img = cv.line(img, tuple(axisPoints[3].ravel()), tuple(axisPoints[2].ravel()), (0,0,255), 3)
+    return img
+
+def draw_cube(img, corners, imgpts):
+    imgpts = np.int32(imgpts).reshape(-1,2)
+    # draw ground floor in green
+    img = cv.drawContours(img, [imgpts[:4]],-1,(0,255,0),-3)
+    # draw pillars in blue color
+    for i,j in zip(range(4),range(4,8)):
+        img = cv.line(img, tuple(imgpts[i]), tuple(imgpts[j]),(255),3)
+
+    # draw top layer in red color
+    img = cv.drawContours(img, [imgpts[4:]],-1,(0,0,255),3)
+    return img
+
+# # Code from https://www.learnopencv.com/rotation-matrix-to-euler-angles/
+# # Calculates rotation matrix to euler angles
+# # The result is the same as MATLAB except the order
+# # of the euler angles ( x and z are swapped ).
+# def rotationMatrixToEulerAngles(R) :
+
+#     sy = np.sqrt(R[0,0] * R[0,0] +  R[1,0] * R[1,0])
+
+#     singular = sy < 1e-6
+
+#     if  not singular :
+#         x = np.arctan2(R[2,1] , R[2,2])
+#         y = np.arctan2(-R[2,0], sy)
+#         z = np.arctan2(R[1,0], R[0,0])
+#     else :
+#         x = np.arctan2(-R[1,2], R[1,1])
+#         y = np.arctan2(-R[2,0], sy)
+#         z = 0
+
+#     theta = np.rad2deg(np.array([x, y, z]))
+#     return theta
+
+
+# # Calculates Rotation Matrix given euler angles.
+# def eulerAnglesToRotationMatrix(theta) :
+
+#     theta = np.deg2rad(theta)
+ 
+#     R_x = np.array([[1,         0,                  0                   ],
+#                     [0,         np.cos(theta[0]), -np.sin(theta[0]) ],
+#                     [0,         np.sin(theta[0]), np.cos(theta[0])  ]
+#                     ])
+ 
+#     R_y = np.array([[np.cos(theta[1]),    0,      np.sin(theta[1])  ],
+#                     [0,                     1,      0                   ],
+#                     [-np.sin(theta[1]),   0,      np.cos(theta[1])  ]
+#                     ])
+ 
+#     R_z = np.array([[np.cos(theta[2]),    -np.sin(theta[2]),    0],
+#                     [np.sin(theta[2]),    np.cos(theta[2]),     0],
+#                     [0,                     0,                      1]
+#                     ])
+ 
+#     R = np.dot(R_z, np.dot( R_y, R_x ))
+ 
+#     return R
+
+#%% Main
+class PlaneDetector:
+    def __init__(self):
+
+        self.frame_size = (640,480)
```

Comment:
must be sure that this resolution is available for the device - is there some check or some exception raised in case it is not available?